kafka/proxy.go

```diff
@@ -109,58 +146,70 @@ func (h *produceRequestHandler) Handle(requestKeyVersion *kafkaprotocol.RequestK
 		return shouldReply, nil
 	}
 
-	msgs := h.extractMessages(req)
+	msgs, err := h.extractMessages(req)
+	if err != nil {
+		logrus.WithError(err).Error("error extracting messages")
+		return shouldReply, nil
+	}
+
 	if len(msgs) == 0 {
 		logrus.Error("The produce request has no messages")
```

Comment:
`Warn` or `Info` will be better than `Error`

Reply:
A Kafka Produce Request **should** contain messages. If there are no messages, something weird happened here. Most probably the proxy is not being able to parse the message correctly?

Perhaps what should be changed is the message to be more explicit: `Unexpected error happened: The produce request has no messages`. 

WDYT?

Reply:
Ok, it can stay :)